src/components/Pressable.tsx

```diff
@@ -0,0 +1,286 @@
+  const isPressCallbackEnabled = useRef<boolean>(true);
+  const isPressedDown = useRef<boolean>(false);
+
+  const normalizedHitSlop: Insets =
+    typeof props.hitSlop === 'number'
+      ? numberAsInset(props.hitSlop)
+      : props.hitSlop ?? {};
+
+  const normalizedPressRetentionOffset: Insets =
+    typeof props.pressRetentionOffset === 'number'
+      ? numberAsInset(props.pressRetentionOffset)
+      : props.pressRetentionOffset ?? {};
+
+  const pressGesture = Gesture.LongPress().onStart((event) => {
+    if (isPressedDown.current) {
+      props.onLongPress?.(adaptStateChangeEvent(event));
+      isPressCallbackEnabled.current = false;
+    }
+  });
+
+  const setPressedState = (isPressed: boolean) => {
+    isPressedDown.current = isPressed;
+    if (typeof props.style === 'function') {
+      setStyleProp(props.style({ pressed: isPressed }));
+    } else {
+      setStyleProp(props.style);
+    }
+    if (typeof props.children === 'function') {
+      setChildrenProp(props.children({ pressed: isPressed }));
+    } else {
+      setChildrenProp(props.children);
+    }
+  };
+
+  const hoverGesture = Gesture.Hover()
+    .onBegin((event) => {
+      setTimeout(
+        () => props.onHoverIn?.(adaptStateChangeEvent(event)),
+        props.delayHoverIn ?? DEFAULT_HOVER_DELAY
+      );
+    })
+    .onEnd((event) => {
+      setTimeout(
+        () => props.onHoverOut?.(adaptStateChangeEvent(event)),
+        props.delayHoverOut ?? DEFAULT_HOVER_DELAY
+      );
+    });
+
+  const touchGesture = Gesture.Manual()
+    .onTouchesDown((event) => {
+      pressableRef.current?.measure((_x, _y, width, height) => {
+        if (
+          !isTouchWithinInset(event.changedTouches.at(-1), normalizedHitSlop, {
+            width,
+            height,
+          }) ||
+          isPressedDown.current
+        ) {
+          return;
+        }
+
+        props.onPressIn?.(adaptTouchEvent(event));
+
+        isPressCallbackEnabled.current = true;
+        setPressedState(true);
+      });
+    })
+    .onTouchesUp((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      if (isPressCallbackEnabled.current) {
+        props.onPress?.(adaptTouchEvent(event));
+      }
+
+      setPressedState(false);
+    })
+    .onTouchesCancelled((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      // original triggers onPressOut on cancel, but not onPress
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      setPressedState(false);
+    });
+
+  const rippleGesture = Gesture.Native();
+
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
```

Comment:
Why do it twice?

Reply:
mistake when i was testing `unstable_pressDelay`